hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -432,6 +468,7 @@ private boolean initializeFromFilesystem(String initializationTime, List<Metadat
       bulkCommit(commitTimeForPartition, partitionType, records, fileGroupCount);
       metadataMetaClient.reloadActiveTimeline();
       dataMetaClient.getTableConfig().setMetadataPartitionState(dataMetaClient, partitionType, true);
+      initMetadataReader();
```

Comment:
add java docs calling out the requirement

Reply:
This is required as we need to update the reader after each partition is initialized. e.g. record index initialization reads the list of partitions from FILES partition. 
     List<String> partitions = metadata.getAllPartitionPaths();

I have removed this line and moved this to after FILES partition is initialized.

Reply:
Let's keep it here itself because this call will init `HoodieBackedTableMetadata` which will check whether MDT is initialized or not based in table config, which get updated only after `dataMetaClient.getTableConfig().setMetadataPartitionState`.